diff --git a/arch/arm64/boot/dts/renesas/r9a07g044.dtsi b/arch/arm64/boot/dts/renesas/r9a07g044.dtsi
index ef09528d316f..c055686d9166 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g044.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a07g044.dtsi
@@ -1181,7 +1181,7 @@ port@1 {
 		};

 		cpg: clock-controller@11010000 {
-			compatible = "renesas,r9a07g044-cpg";
+			compatible = "renesas,r9a07g044-cpg", "syscon";
 			reg = <0 0x11010000 0 0x10000>;
 			clocks = <&extal_clk>;
 			clock-names = "extal";
@@ -1191,7 +1191,7 @@ cpg: clock-controller@11010000 {
 		};

 		sysc: system-controller@11020000 {
-			compatible = "renesas,r9a07g044-sysc";
+			compatible = "renesas,r9a07g044-sysc", "syscon";
 			reg = <0 0x11020000 0 0x10000>;
 			interrupts = <GIC_SPI 42 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 43 IRQ_TYPE_LEVEL_HIGH>,
diff --git a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
index 408c7ca47350..02980a35079a 100644
--- a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
@@ -53,6 +53,45 @@ mmp_reserved: linux,multimedia {
 			reg = <0x0 0x68000000 0x0 0x8000000>;
 		};

+		vdev0vring0: vdev0vring0@43000000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x43000000 0x0 0x50000>;
+			no-map;
+		};
+
+		vdev0vring1: vdev0vring1@43050000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x43050000 0x0 0x50000>;
+			no-map;
+		};
+
+		vdev0buffer: vdev0buffer@43200000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x43200000 0x0 0x300000>;
+			no-map;
+		};
+	};
+
+	cm33_rproc: cm33 {
+		compatible = "renesas,rz-cm33";
+		reg = <0x0 0x00010000 0x0 0x0000800>,
+				<0x0 0x0001FF80 0x0 0x0000080>,
+				<0x0 0x40010000 0x0 0x3EF0000>;
+		reg-names = "cm33_sram_ns", "cm33_sram_s", "cm33_ddr";
+		memory-region = <&vdev0vring0>, <&vdev0vring1>, <&vdev0buffer>;
+		renesas,rz-cpg = <&cpg>;
+		renesas,rz-sysc = <&sysc>;
+		renesas,rz-bootaddrs = <0x1001FF80>, <0x00010000>;
+		renesas,rz-rsctbl = <0x42F00000>;
+		clocks = <&cpg CPG_MOD R9A07G044_CM33_CLKIN>,
+				<&cpg CPG_MOD R9A07G044_CM33_TSCLK>;
+		clock-names = "clkin", "tsclk";
+		resets = <&cpg R9A07G044_CM33_NPORESET>,
+				<&cpg R9A07G044_CM33_NSYSRESET>,
+				<&cpg R9A07G044_CM33_MISCRESETN>;
+		reset-names = "nporeset", "nsysreset", "miscresetn";
+		power-domains = <&cpg>;
+		status = "okay";
 	};

 	mmngr {
