diff --git a/drivers/clk/renesas/r9a07g043-cpg.c b/drivers/clk/renesas/r9a07g043-cpg.c
index 8d9ac7f4338e..02c0f08296ec 100644
--- a/drivers/clk/renesas/r9a07g043-cpg.c
+++ b/drivers/clk/renesas/r9a07g043-cpg.c
@@ -222,6 +222,10 @@ static struct rzg2l_mod_clk r9a07g043_mod_clks[] = {
 	DEF_MOD("lcdc_clk_d",	R9A07G043_LCDC_CLK_D, R9A07G043_CLK_M3,
 				0x56c, 1),
 #endif
+	DEF_MOD("cm33_clkin",   R9A07G043_CM33_CLKIN, R9A07G043_CLK_I2,
+				0x504, 0),
+	DEF_MOD("cm33_tsclk",   R9A07G043_CM33_TSCLK, R9A07G043_OSCCLK,
+				0x504, 1),
 	DEF_MOD("ssi0_pclk",	R9A07G043_SSI0_PCLK2, R9A07G043_CLK_P0,
 				0x570, 0),
 	DEF_MOD("ssi0_sfr",	R9A07G043_SSI0_PCLK_SFR, R9A07G043_CLK_P0,
@@ -355,6 +359,9 @@ static struct rzg2l_reset r9a07g043_resets[] = {
 	DEF_RST(R9A07G043_ADC_PRESETN, 0x8a8, 0),
 	DEF_RST(R9A07G043_ADC_ADRST_N, 0x8a8, 1),
 	DEF_RST(R9A07G043_TSU_PRESETN, 0x8ac, 0),
+	DEF_RST(R9A07G043_CM33_NPORESET, 0x804, 0),
+	DEF_RST(R9A07G043_CM33_NSYSRESET, 0x804, 1),
+	DEF_RST(R9A07G043_CM33_MISCRESETN, 0x804, 2),
 #ifdef CONFIG_RISCV
 	DEF_RST(R9A07G043_NCEPLIC_ARESETN, 0x908, 0),
 #endif
@@ -396,7 +403,7 @@ const struct rzg2l_cpg_info r9a07g043_cpg_info = {
 	.mod_clks = r9a07g043_mod_clks,
 	.num_mod_clks = ARRAY_SIZE(r9a07g043_mod_clks),
 #ifdef CONFIG_ARM64
-	.num_hw_mod_clks = R9A07G043_TSU_PCLK + 1,
+	.num_hw_mod_clks = R9A07G043_CM33_TSCLK + 1,
 
 	/* No PM Module Clocks */
 	.no_pm_mod_clks = r9a07g043_no_pm_mod_clks,
@@ -409,7 +416,7 @@ const struct rzg2l_cpg_info r9a07g043_cpg_info = {
 	/* Resets */
 	.resets = r9a07g043_resets,
 #ifdef CONFIG_ARM64
-	.num_resets = R9A07G043_TSU_PRESETN + 1, /* Last reset ID + 1 */
+	.num_resets = R9A07G043_CM33_MISCRESETN + 1, /* Last reset ID + 1 */
 #endif
 #ifdef CONFIG_RISCV
 	.num_resets = R9A07G043_IAX45_RESETN + 1, /* Last reset ID + 1 */
diff --git a/include/dt-bindings/clock/r9a07g043-cpg.h b/include/dt-bindings/clock/r9a07g043-cpg.h
index 77cde8effdc7..5dcc926b85e1 100644
--- a/include/dt-bindings/clock/r9a07g043-cpg.h
+++ b/include/dt-bindings/clock/r9a07g043-cpg.h
@@ -108,15 +108,17 @@
 #define R9A07G043_ADC_ADCLK		76
 #define R9A07G043_ADC_PCLK		77
 #define R9A07G043_TSU_PCLK		78
-#define R9A07G043_NCEPLDM_DM_CLK	79	/* RZ/Five Only */
-#define R9A07G043_NCEPLDM_ACLK		80	/* RZ/Five Only */
-#define R9A07G043_NCEPLDM_TCK		81	/* RZ/Five Only */
-#define R9A07G043_NCEPLMT_ACLK		82	/* RZ/Five Only */
-#define R9A07G043_NCEPLIC_ACLK		83	/* RZ/Five Only */
-#define R9A07G043_AX45MP_CORE0_CLK	84	/* RZ/Five Only */
-#define R9A07G043_AX45MP_ACLK		85	/* RZ/Five Only */
-#define R9A07G043_IAX45_CLK		86	/* RZ/Five Only */
-#define R9A07G043_IAX45_PCLK		87	/* RZ/Five Only */
+#define R9A07G043_CM33_CLKIN        79
+#define R9A07G043_CM33_TSCLK        80
+#define R9A07G043_NCEPLDM_DM_CLK	81	/* RZ/Five Only */
+#define R9A07G043_NCEPLDM_ACLK		82	/* RZ/Five Only */
+#define R9A07G043_NCEPLDM_TCK		83	/* RZ/Five Only */
+#define R9A07G043_NCEPLMT_ACLK		84	/* RZ/Five Only */
+#define R9A07G043_NCEPLIC_ACLK		85	/* RZ/Five Only */
+#define R9A07G043_AX45MP_CORE0_CLK	86	/* RZ/Five Only */
+#define R9A07G043_AX45MP_ACLK		87	/* RZ/Five Only */
+#define R9A07G043_IAX45_CLK		    88	/* RZ/Five Only */
+#define R9A07G043_IAX45_PCLK		89	/* RZ/Five Only */
 
 /* R9A07G043 Resets */
 #define R9A07G043_CA55_RST_1_0		0	/* RZ/G2UL Only */
@@ -189,16 +191,19 @@
 #define R9A07G043_ADC_PRESETN		67
 #define R9A07G043_ADC_ADRST_N		68
 #define R9A07G043_TSU_PRESETN		69
-#define R9A07G043_NCEPLDM_DTM_PWR_RST_N	70	/* RZ/Five Only */
-#define R9A07G043_NCEPLDM_ARESETN	71	/* RZ/Five Only */
-#define R9A07G043_NCEPLMT_POR_RSTN	72	/* RZ/Five Only */
-#define R9A07G043_NCEPLMT_ARESETN	73	/* RZ/Five Only */
-#define R9A07G043_NCEPLIC_ARESETN	74	/* RZ/Five Only */
-#define R9A07G043_AX45MP_ARESETNM	75	/* RZ/Five Only */
-#define R9A07G043_AX45MP_ARESETNS	76	/* RZ/Five Only */
-#define R9A07G043_AX45MP_L2_RESETN	77	/* RZ/Five Only */
-#define R9A07G043_AX45MP_CORE0_RESETN	78	/* RZ/Five Only */
-#define R9A07G043_IAX45_RESETN		79	/* RZ/Five Only */
+#define R9A07G043_CM33_NPORESET         70
+#define R9A07G043_CM33_NSYSRESET        71
+#define R9A07G043_CM33_MISCRESETN       72
+#define R9A07G043_NCEPLDM_DTM_PWR_RST_N	73	/* RZ/Five Only */
+#define R9A07G043_NCEPLDM_ARESETN	    74	/* RZ/Five Only */
+#define R9A07G043_NCEPLMT_POR_RSTN	    75	/* RZ/Five Only */
+#define R9A07G043_NCEPLMT_ARESETN	    76	/* RZ/Five Only */
+#define R9A07G043_NCEPLIC_ARESETN	    77	/* RZ/Five Only */
+#define R9A07G043_AX45MP_ARESETNM	    78	/* RZ/Five Only */
+#define R9A07G043_AX45MP_ARESETNS	    79	/* RZ/Five Only */
+#define R9A07G043_AX45MP_L2_RESETN	    80	/* RZ/Five Only */
+#define R9A07G043_AX45MP_CORE0_RESETN	81	/* RZ/Five Only */
+#define R9A07G043_IAX45_RESETN		    82	/* RZ/Five Only */
 
 
 #endif /* __DT_BINDINGS_CLOCK_R9A07G043_CPG_H__ */
