diff --git a/arch/arm64/boot/dts/renesas/r9a07g043.dtsi b/arch/arm64/boot/dts/renesas/r9a07g043.dtsi
index 90e7ab466a8b..776192240135 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g043.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a07g043.dtsi
@@ -68,6 +68,30 @@ opp-1000000000 {
 		};
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		vdev0vring0: vdev0vring0@0x43000000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x43000000 0x0 0x50000>;
+			no-map;
+		};
+
+		vdev0vring1: vdev0vring1@0x43050000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x43050000 0x0 0x50000>;
+			no-map;
+		};
+
+		vdev0buffer: vdev0buffer@0x43200000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x43200000 0x0 0x300000>;
+			no-map;
+		};
+	};
+
 	soc: soc {
 		compatible = "simple-bus";
 		#address-cells = <2>;
@@ -143,6 +167,28 @@ mtu3: timer@10001200 {
 			status = "disabled";
 		};
 
+		cm33_rproc: cm33 {
+			compatible = "renesas,rz-cm33";
+			reg = <0x0 0x00010000 0x0 0x0000800>,
+			      <0x0 0x0001FF80 0x0 0x0000080>,
+			      <0x0 0x40010000 0x0 0x3EF0000>;
+			reg-names = "cm33_sram_ns", "cm33_sram_s", "cm33_ddr";
+			memory-region = <&vdev0vring0>, <&vdev0vring1>, <&vdev0buffer>;
+			renesas,rz-cpg = <&cpg>;
+			renesas,rz-sysc = <&sysc>;
+			renesas,rz-bootaddrs = <0x1001FF80>, <0x00010000>;
+			renesas,rz-rsctbl = <0x42F00000>;
+			clocks = <&cpg CPG_MOD R9A07G043_CM33_CLKIN>,
+				 <&cpg CPG_MOD R9A07G043_CM33_TSCLK>;
+			clock-names = "clkin", "tsclk";
+			resets = <&cpg R9A07G043_CM33_NPORESET>,
+				 <&cpg R9A07G043_CM33_NSYSRESET>,
+				 <&cpg R9A07G043_CM33_MISCRESETN>;
+			reset-names = "nporeset", "nsysreset", "miscresetn";
+			power-domains = <&cpg>;
+			status = "okay";
+		};
+
 		poe3: poe3@10049800 {
 			compatible = "renesas,r9a07g043-poe3",
 				     "renesas,rz-poe3";
