diff --git a/arch/arm64/boot/dts/renesas/r9a09g077m44-dev.dts b/arch/arm64/boot/dts/renesas/r9a09g077m44-dev.dts
index 550733a328e1..1206dc729c32 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g077m44-dev.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g077m44-dev.dts
@@ -127,6 +127,12 @@ global_cma: linux,cma@D0000000 {
 			reusable;
 			reg = <0x0 0xD0000000 0x0 0x10000000>;
 		};
+		
+		ca55ddr: ca55ddr@C6000000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0xC6000000 0x0 0x900000>;
+			no-map;
+		};		
 	};
 
 	reg_1p8v: regulator0 {
-- 
2.34.1

