From fbbeb12229089e8cb55441ddb017da839ec15aaf Mon Sep 17 00:00:00 2001
From: Trung Le <trung.le.uj@renesas.com>
Date: Thu, 5 Feb 2026 14:58:43 +0700
Subject: [PATCH 1/9] Set SCIF1 and OSTM1 OSTM2 as critical clock

---
 drivers/clk/renesas/r9a08g045-cpg.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/clk/renesas/r9a08g045-cpg.c b/drivers/clk/renesas/r9a08g045-cpg.c
index 6101a304562d..743bef60df9e 100644
--- a/drivers/clk/renesas/r9a08g045-cpg.c
+++ b/drivers/clk/renesas/r9a08g045-cpg.c
@@ -381,6 +381,9 @@ static const unsigned int r9a08g045_crit_mod_clks[] __initconst = {
 	MOD_CLK_BASE + R9A08G045_IA55_CLK,
 	MOD_CLK_BASE + R9A08G045_DMAC_ACLK,
 	MOD_CLK_BASE + R9A08G045_VBAT_BCLK,
+	MOD_CLK_BASE + R9A08G045_SCIF1_CLK_PCK,
+	MOD_CLK_BASE + R9A08G045_OSTM1_PCLK,
+	MOD_CLK_BASE + R9A08G045_OSTM2_PCLK,
 };
 
 static const unsigned int r9a08g045_no_pm_mod_clks[] = {
-- 
2.34.1

