diff --git a/arch/arm64/boot/dts/renesas/r9a08g045.dtsi b/arch/arm64/boot/dts/renesas/r9a08g045.dtsi
index 132b79389ea7..18dbbbf1b6d7 100644
--- a/arch/arm64/boot/dts/renesas/r9a08g045.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a08g045.dtsi
@@ -97,6 +97,31 @@ psci {
 		method = "smc";
 	};

+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		vdev0vring0: vdev0vring0@0x43000000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x43000000 0x0 0x50000>;
+			no-map;
+		};
+
+		vdev0vring1: vdev0vring1@0x43050000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x43050000 0x0 0x50000>;
+			no-map;
+		};
+
+		vdev0buffer: vdev0buffer@0x43200000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x43200000 0x0 0x300000>;
+			no-map;
+		};
+	};
+
 	soc: soc {
 		compatible = "simple-bus";
 		interrupt-parent = <&gic>;
@@ -1481,6 +1506,49 @@ tsu: thermal@10059400 {
 			adc-handle = <&adc>;
 			#thermal-sensor-cells = <1>;
 		};
+		cm33_rproc: cm33 {
+			compatible = "renesas,rz-cm33";
+			reg = <0x0 0x00020000 0x0 0x003FFFF>,
+			      <0x0 0x40010000 0x0 0x3EF0000>;
+			reg-names = "cm33_sram", "cm33_ddr";
+			memory-region = <&vdev0vring0>, <&vdev0vring1>, <&vdev0buffer>;
+			renesas,rz-core = <0x0>;
+			renesas,rz-cpg = <&cpg>;
+			renesas,rz-sysc = <&sysc>;
+			renesas,rz-bootaddrs = <0x00023000>, <0x10023800>;
+			renesas,rz-rsctbl = <0x42F00000>;
+			clocks = <&cpg CPG_MOD R9A08G045_CM33_CLKIN>,
+				 <&cpg CPG_MOD R9A08G045_CM33_TSCLK>;
+			clock-names = "clkin", "tsclk";
+			resets = <&cpg R9A08G045_CM33_NPORESET>,
+				 <&cpg R9A08G045_CM33_NSYSRESET>,
+				 <&cpg R9A08G045_CM33_MISCRESETN>;
+			reset-names = "nporeset", "nsysreset", "miscresetn";
+			power-domains = <&cpg>;
+			status = "okay";
+		};
+
+		cm33_fpu_rproc: cm33_fpu {
+			compatible = "renesas,rz-cm33";
+			reg = <0x0 0x00060000 0x0 0x003FFFF>,
+			      <0x0 0x40010000 0x0 0x3EF0000>;
+			reg-names = "cm33_sram", "cm33_ddr";
+			memory-region = <&vdev0vring0>, <&vdev0vring1>, <&vdev0buffer>;
+			renesas,rz-core = <0x1>;
+			renesas,rz-cpg = <&cpg>;
+			renesas,rz-sysc = <&sysc>;
+			renesas,rz-bootaddrs = <0x00060000>, <0x10060800>;
+			renesas,rz-rsctbl = <0x42F00000>;
+			clocks = <&cpg CPG_MOD R9A08G045_CM33_FPU_CLKIN>,
+				 <&cpg CPG_MOD R9A08G045_CM33_FPU_TSCLK>;
+			clock-names = "clkin", "tsclk";
+			resets = <&cpg R9A08G045_CM33_FPU_NPORESET>,
+				 <&cpg R9A08G045_CM33_FPU_NSYSRESET>,
+				 <&cpg R9A08G045_CM33_FPU_MISCRESETN>;
+			reset-names = "nporeset", "nsysreset", "miscresetn";
+			power-domains = <&cpg>;
+			status = "okay";
+		};

 		xspi: spi@10060000 {
 			compatible = "renesas,g3s-xspi-if";
