diff --git a/plat/renesas/rz/soc/g3s/include/rz_soc_def.h b/plat/renesas/rz/soc/g3s/include/rz_soc_def.h
index 67617c7b8..047e6e346 100644
--- a/plat/renesas/rz/soc/g3s/include/rz_soc_def.h
+++ b/plat/renesas/rz/soc/g3s/include/rz_soc_def.h
@@ -71,7 +71,11 @@
  * 0x64000 has been chosen as FIP base as that leaves room at start of xSPI Flash
  * to store a 128K BL2 image.
  */
+#ifdef ENABLE_RZG3S_CM33_BOOT
+#define RZG3S_BL2_SIZE_MAX				UL(0x00260000)
+#else
 #define RZG3S_BL2_SIZE_MAX				UL(0x00060000)
+#endif
 #define RZG3S_FIP_SIZE_MAX				UL(0x001A0000)
 #define RZG3S_FIP_LIMIT					(RZG3S_SPIROM_BASE + RZG3S_FIP_SIZE_MAX)
 
@@ -79,7 +83,12 @@
 #define RZG3S_EMMC_FIP_BASE				(RZG3S_BL2_SIZE_MAX)
 #define RZG3S_SD_FIP_BASE				(RZG3S_BL2_SIZE_MAX)
 
+#ifdef ENABLE_RZG3S_CM33_BOOT
+#define RZG3S_M33_FW_OFFSET				UL(0x00400000)
+#else
 #define RZG3S_M33_FW_OFFSET				UL(0x00200000)
+#endif
+
 #define RZG3S_SPIROM_M33_FW_BASE		(RZG3S_SPIROM_BASE + RZG3S_M33_FW_OFFSET)
 #define RZG3S_EMMC_M33_FW_BASE			(RZG3S_M33_FW_OFFSET)
 #define RZG3S_SD_M33_FW_BASE			(RZG3S_M33_FW_OFFSET)
diff --git a/plat/renesas/rz/soc/g3s/platform.mk b/plat/renesas/rz/soc/g3s/platform.mk
index 9a68beaa7..3769d4414 100644
--- a/plat/renesas/rz/soc/g3s/platform.mk
+++ b/plat/renesas/rz/soc/g3s/platform.mk
@@ -46,6 +46,9 @@ BL31_SOURCES			+=	plat/renesas/rz/soc/g3s/bl31_plat_setup.c				\
 ifneq (${ENABLE_STACK_PROTECTOR},0)
 PLAT_BL_COMMON_SOURCES	+=	plat/renesas/rz/soc/g3s/rz_stack_protector.c
 endif
+ifeq (${ENABLE_RZG3S_CM33_BOOT}, 1)
+$(eval $(call add_define, ENABLE_RZG3S_CM33_BOOT))
+endif
 
 ifneq (${TRUSTED_BOARD_BOOT},0)
 	PLAT_BL_COMMON_SOURCES	+=	plat/renesas/rz/soc/g3s/drivers/auth/cip/cip.c
