diff --git a/board/renesas/rzg3l-dev/rzg3l-dev.c b/board/renesas/rzg3l-dev/rzg3l-dev.c
index 2e9fedd16d..5ef50e475c 100644
--- a/board/renesas/rzg3l-dev/rzg3l-dev.c
+++ b/board/renesas/rzg3l-dev/rzg3l-dev.c
@@ -123,6 +123,11 @@ DECLARE_GLOBAL_DATA_PTR;
 #define LPSTS			0x102
 #define	AHB_BUS_CTR		0x208
 
+/* SYSC */
+#define SYSC_BASE			0x11020000
+#define SYS_LSI_MODE		(SYSC_BASE + 0x0A00)
+#define SYS_STAT_DEBUGEN_MASK	(1 << 9)
+
 void s_init(void)
 {
 	/* Disable GPIO Write Protect */
@@ -134,16 +139,17 @@ void s_init(void)
 	*(volatile u8 *)(PFC_PMC30) |= 0x3F;	/* PG0-PG5 function	*/
 
 #if CONFIG_TARGET_SMARC_RZG3L
-	/* PJ1 = 0 QSD1_IOVS=0	*/
+	if ((*(volatile u32 *)(SYS_LSI_MODE) & SYS_STAT_DEBUGEN_MASK) == 0) {
+		/* PJ1 = 0 QSD1_IOVS=0	*/
 	
-	*(volatile u8 *)(PFC_PMC33) &= ~(0x02);
-	*(volatile u16 *)(PFC_PM33) = (*(volatile u16 *)(PFC_PM33) & 0xFF) | 0x08; /*bit[5:4]='B10	*/
-	*(volatile u8 *)(PFC_P33) &= ~(0x02);
-	/* PJ2 = 1 QSD1_PWEN=1	*/
-	*(volatile u8 *)(PFC_PMC33) &= ~(0x04);
-	*(volatile u16 *)(PFC_PM33) = (*(volatile u16 *)(PFC_PM33) & 0xFF) | 0x20; /*bit[5:4]='B10	*/
-	*(volatile u8 *)(PFC_P33) |= 0x04;
-
+		*(volatile u8 *)(PFC_PMC33) &= ~(0x02);
+		*(volatile u16 *)(PFC_PM33) = (*(volatile u16 *)(PFC_PM33) & 0xFF) | 0x08; /*bit[5:4]='B10	*/
+		*(volatile u8 *)(PFC_P33) &= ~(0x02);
+		/* PJ2 = 1 QSD1_PWEN=1	*/
+		*(volatile u8 *)(PFC_PMC33) &= ~(0x04);
+		*(volatile u16 *)(PFC_PM33) = (*(volatile u16 *)(PFC_PM33) & 0xFF) | 0x20; /*bit[5:4]='B10	*/
+		*(volatile u8 *)(PFC_P33) |= 0x04;
+	}
 #else
 	/* P56 = 1 QSD1_PWEN=1	*/
 	*(volatile u8 *)(PFC_PMC25) &= ~(0x40);
